addons/point_of_sale/static/src/js/models.js

```diff
@@ -382,10 +382,15 @@ exports.PosModel = Backbone.Model.extend({
                     return a.id - b.id;
                 }
             });
-            self.payment_methods_by_id = {}
+            self.payment_methods_by_id = {};
             _.each(self.payment_methods, function(payment_method) {
-                self.payment_methods_by_id[payment_method.id] = payment_method
-            })
+                self.payment_methods_by_id[payment_method.id] = payment_method;
+
+                var payment_interface = self.electronic_payment_methods[payment_method.use_payment_terminal];
+                if (payment_interface) {
+                    payment_method.payment_terminal = new payment_interface(self, payment_method);
```

Comment:
isn't it new PaymentInterface instead of new payment_interface and add require at begining of the file?

Reply:
No, it's confusing because I didn't name the variables correctly. I renamed `self.electronic_payment_methods` to `self.electronic_payment_interfaces` and used CamelCase where appropriate to imply they are classes. I also added some documentation to `register_payment_method` to clarify further.

In summary, `self.electronic_payment_interfaces` will be populated by implementations like `pos_adyen` and `pos_iot`. They'll add their implemented classes which inherit from PaymentInterface via `register_payment_method`. The code above will instantiate the right interface per payment method.